pkg/docker/docker.go

```diff
@@ -82,6 +83,16 @@ func (d *Docker) baseRun(out io.Writer, entrypoint string, args []string, env ..
 		"-e", "KUBECONFIG=" + harness.DefaultKubeConfig,
 		"--dns", "1.1.1.1",
 		"--entrypoint", entrypoint,
+		"--cap-add", "NET_ADMIN",
+	}
+
+	// add-host entries, the ADD_HOSTS env var will look like:
+	// host1:ip1,host2:ip2...
+	addHostEntries := strings.Split(os.Getenv("ADD_HOSTS"), ",")
+	for _, entry := range addHostEntries {
+		if entry != "" {
+			baseArgs = append(baseArgs, "--add-host", entry)
+		}
 	}
 
```

Comment:
How iptables are set?

Reply:
iptables are installed in the image and the `e2e-harness` user (the default) is allowed to call it as root. It's up to the consumer projects to use the command as needed.